core/src/worker/workflow/mod.rs

```diff
@@ -210,13 +210,25 @@ impl Workflows {
         &self,
         completion: WorkflowActivationCompletion,
     ) -> Result<usize, CompleteWfError> {
+        let is_empty_completion = completion.is_empty();
```

Comment:
Empty completion doesn't necessarily imply eviction response (as you obviously know).